source/Octopus.TestPortForwarder/PortForwarder.cs

```diff
@@ -48,13 +49,15 @@ public class PortForwarder : IDisposable
             var scheme = originServer.Scheme;
 
             Start();
-
-            ListeningPort = ((IPEndPoint)listeningSocket.LocalEndPoint).Port;
+            var ipEndPoint = listeningSocket.LocalEndPoint as IPEndPoint ?? throw new InvalidOperationException("listeningSocket.LocalEndPoint was not an IPEndPoint");
+            
+            ListeningPort = ipEndPoint.Port;
             PublicEndpoint = new UriBuilder(scheme, "localhost", ListeningPort).Uri;
 
             Task.Factory.StartNew(() => WorkerTask(cancellationTokenSource.Token).ConfigureAwait(false), TaskCreationOptions.LongRunning);
         }
 
+        [MemberNotNull(nameof(listeningSocket))]
```

Comment:
This tells the code in the constructor that `Start()` is guaranteed to initialize `listeningSocket` so it knows it won't be null there